design-spec/priviledge-decentralized-swupdates.tex

```diff
@@ -90,33 +90,85 @@ \section{Requirements}
 \label{sec:requirements}
 % FROM: 2 Requirements for a decentralized software update system
 
-This section presents the requirements for achieving decentralized governance of
-a blockchain.
+This section presents the requirements of an update mechanism for achieving
+decentralized governance of a blockchain.
 
 \subsection{Open}
 \label{sec:open-participation}
 
+In the absence of malicious actors, we could aspire that anybody who can
+participate the blockchain protocol by submitting transactions can participate
+in the update process. However allowing this would open the door for spam
+attacks: if the barrier for submitting update proposals is as low as submitting
+regular transactions, then we risk having a large number of bogus update
+proposals being submitted that will overwhelm the human actors that have to
+review these proposals.
+
+In light of the risk presented in the preceding paragraph, we require that the
+group of people that can participate in the update process should be as broad as
+possible, but not broader. This means that:
+\begin{itemize}
+\item Restrictions on the group of people that can submit update proposals
+  should be as loose as possible.
+\item Any participant that can submit a common transaction should be able to
+  vote on update proposals.
+\end{itemize}
+
 \subsection{Democratic}
 \label{sec:decentr-decis-making}
 
+All major decisions in the life cycle of a system update should be made by the
+community, via a secure voting mechanism.
+
 \subsection{Protocol driven}
 \label{sec:protocol-driven}
 
+The update protocol should be an on-chain process, which is automatically
+enforced by the ledger rules of the blockchain.
+
 \subsection{Transparent and auditable}
 \label{sec:transp-audit}
 
+Anybody should be able to answer \emph{when}, \emph{why}, and \emph{how} the
+system has evolved the way it has. The systems' update history should be:
+\begin{itemize}
+\item open to everyone, and
+\item stored in a tamper-free global ledger.
+\end{itemize}
+
 \subsection{Secure}
 \label{sec:secure}
 
+% TODO: what about other security aspects such as: preservation of ownership,
+% resiliency to DoS attacks, replay protection, ensuring participation of an
+% honest majority of stake-holders, ensuring liveness, etc.
+
```

Comment:
Here we are just stating the requirement for a secure update mechanism. This requirement is not only about providing such a mechanism but first and foremost to define what secure means in this context. So in this paragraph, we should only mention that we have to define secure updates and then propose a secure solution wrt to this definition.

In the following text we deal with the following security aspects:
- secure activation --> safe threshold that ensures honest majority (threshold analysis) + secure activation protocols
- secure voting  (safety/liveneness) --> threshold analysis
- DoS attacks -> fee-based transactions
- Replay attacks --> delegation certificates are bound to utxos (see delegation section)